addons/web/static/src/core/template_inheritance.js

```diff
@@ -0,0 +1,244 @@
+                                }
+                            }
+                            source = operationContent.cloneNode(true);
+                            if (target.hasAttribute("t-name")) {
+                                source.setAttribute("t-name", target.getAttribute("t-name"));
+                            }
+                            if (comment) {
+                                source.prepend(comment);
+                            }
+                        }
+                        break;
+                    }
+                    case "inner":
+                        while (target.firstChild) {
+                            target.removeChild(target.lastChild);
+                        }
+                        target.append(...operation.childNodes);
+                        break;
+                    default:
+                        throw new Error(`Invalid mode attribute: '${mode}'`);
+                }
+                break;
+            }
+            case "attributes": {
+                for (const child of operation.children) {
+                    if (child.tagName !== "attribute") {
+                        continue;
+                    }
+                    const attributeName = child.getAttribute("name");
+                    const firstNode = child.childNodes[0];
+                    let value = firstNode?.nodeType === Node.TEXT_NODE ? firstNode.data : "";
+
+                    if (child.hasAttribute("add") || child.hasAttribute("remove")) {
+                        if (firstNode?.nodeType === Node.TEXT_NODE) {
+                            throw new Error();
+                        }
+                        const separator = child.getAttribute("separator", ",");
+                        const toRemove = new Set();
+                        for (const s of (child.getAttribute("remove") || "").split(separator)) {
+                            toRemove.add(s.trim());
+                        }
+                        const values = [];
+                        for (const s of (target.getAttribute(attributeName) || "").split(
+                            separator
+                        )) {
+                            const trimmed = s.trim();
+                            if (!toRemove.has(trimmed)) {
+                                values.push(trimmed);
+                            }
+                        }
+                        for (const s of (child.getAttribute("add") || "").split(separator)) {
+                            const trimmed = s.trim();
+                            if (trimmed) {
+                                values.push(trimmed);
+                            }
+                        }
+                        value = values.join(separator || " ");
+                    }
+
+                    if (value) {
+                        target.setAttribute(attributeName, value);
+                    } else if (target.hasAttribute(attributeName)) {
```

Comment:
no need for this if, you can call removeAttribute even if the attribute isn't actually present

Reply:
done